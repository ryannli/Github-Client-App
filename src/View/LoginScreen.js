import React, {
	Component
} from 'react';
import {
	StyleSheet,
	Dimensions,
	Alert,
	SafeAreaView,
	AsyncStorage
} from 'react-native';
import {
	Container,
	Header,
	Content,
	Form,
	Item,
	Input,
	Button,
	Text,
	Body
} from 'native-base';
import {
	TextInput
} from 'react-native';
import Zocial from 'react-native-vector-icons/Zocial';
import api from '../Utils/basicApi';

export default class LoginScreen extends Component {
	state = {}

	/** Handle mouse click event of login. */
	handlePress() {
		if (this.state.username && this.state.password) {
			// Navigate to Tab screen
			var username = this.state.username;
			api.login(this.state.username, this.state.password)
				.then((res => {
					const isValid = res.status < 400;
					if (isValid) {
						AsyncStorage.clear();
						AsyncStorage.setItem('@githubClient:ownername', JSON.stringify(username));
						AsyncStorage.setItem('@githubClient:ownertoken', JSON.stringify(res.data.token));
						this.props.navigation.navigate('TabScreen', {
							'username': username,
							'token': res.data.token,
							'ownername': username,
							'ownertoken': res.data.token,
						});
					}
				})).catch(function(e) {
					Alert.alert(
						'Login Failure',
						'Username and password do not match', );
				});

		} else if (!this.state.username) {
			// The input username cannot be empty
			Alert.alert(
				'Empty Username',
				'Please input a username', );
		} else {
			Alert.alert(
				'Empty Password',
				'Please input a password', );
		}
	}

	/** Render function. */
	render() {
		return (
			<SafeAreaView style={styles.safeArea}>
		      <Container style={styles.container}>
		        <Body style={styles.image}>
		        <Zocial name={'github'} size={px2dpH(225)} />
		        </Body>
		          <Form style={styles.contentInput}>
		            <Item style={styles.input} >
		              <Input placeholder="Username" 
		              		placeholderTextColor="rgba(255,255,255,0.7)" 
		              		onChangeText={(text) => this.setState({'username': text})}
		              		/>
		            </Item>
		            <Item style={styles.input}>
		              <Input placeholder="Password" 
		              		placeholderTextColor="rgba(255,255,255,0.7)" 
		              		onChangeText={(text) => this.setState({'password': text})} 
		              		secureTextEntry />
		            </Item>
		            <Button block style={styles.buttonContainer} onPress={() => this.handlePress()}>
		            	<Text style={styles.buttonText}>Sign In</Text>
		            </Button>
		          </Form>
		          
		      </Container>
		   </SafeAreaView>
		);
	}

}

const deviceW = Dimensions.get('window').width
const deviceH = Dimensions.get('window').height

const basePxW = 375
const basePxH = 750

// Convert to relative width
function px2dpW(px) {
	return px * deviceW / basePxW
}

// Convert to relative height
function px2dpH(px) {
	return px * deviceH / basePxH
}

// Define style sheets
const styles = StyleSheet.create({
	container: {
		backgroundColor: "#03A9F4",
	},
	content: {
		marginBottom: px2dpH(150),
	},
	contentInput: {
		flex: 2,
		marginLeft: 10,
		marginRight: 10,
		marginBottom: 10,
	},
	input: {
		height: px2dpH(60),
		marginBottom: px2dpH(15),
	},
	image: {
		justifyContent: 'center',
		alignItems: 'center',
		flex: 3,
		height: px2dpH(750),
		marginTop: px2dpH(45),
		marginBottom: px2dpH(45),
	},
	buttonContainer: {
		backgroundColor: "#0277BD",
		marginTop: 10,
		marginLeft: 10,
		marginRight: 10,
	},
	buttonText: {
		textAlign: "center",
	},
	safeArea: {
		flex: 1,
		backgroundColor: '#03A9F4',
	}
})