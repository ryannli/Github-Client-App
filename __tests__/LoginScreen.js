import 'react-native';
import React from 'react';
import LoginScreen from '../src/View/LoginScreen';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

test('renders correctly with no password', () => {
	const navigation = {
		state: {
			params: {
				username: "ryannli"
			}
		}
	};
	const tree = renderer.create(
		<LoginScreen navigation={navigation}/>
	);
	expect(tree).toMatchSnapshot();
});

test('renders correctly with no username', () => {
	const navigation = {
		state: {
			params: {
				password: "ryannli"
			}
		}
	};
	const tree = renderer.create(
		<LoginScreen navigation={navigation}/>
	);
	expect(tree).toMatchSnapshot();
});