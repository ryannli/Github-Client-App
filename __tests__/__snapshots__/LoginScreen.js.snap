// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with no password 1`] = `
<RCTSafeAreaView
  style={
    Object {
      "backgroundColor": "#03A9F4",
      "flex": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "flex": 1,
          "height": 1334,
        },
        Object {
          "backgroundColor": "#03A9F4",
        },
      ]
    }
    virtual={undefined}
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "alignSelf": "center",
            "flex": 1,
          },
          Object {
            "alignItems": "center",
            "flex": 3,
            "height": 1334,
            "justifyContent": "center",
            "marginBottom": 80.04,
            "marginTop": 80.04,
          },
        ]
      }
      virtual={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": undefined,
              "fontSize": 400.2,
            },
            undefined,
            Object {
              "fontFamily": "zocial",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
          ]
        }
      >
        
      </Text>
    </View>
    <View
      style={
        Array [
          Object {},
          Object {
            "flex": 2,
            "marginBottom": 10,
            "marginLeft": 10,
            "marginRight": 10,
          },
        ]
      }
      virtual={undefined}
    >
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        collapsable={undefined}
        hasTVPreferredFocus={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "borderBottomWidth": 1,
            "borderColor": "#D9D5DC",
            "borderLeftWidth": 0,
            "borderRightWidth": 0,
            "borderTopWidth": 0,
            "flexDirection": "row",
            "height": 106.72,
            "marginBottom": 26.68,
            "marginLeft": 15,
            "opacity": 1,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <TextInput
          allowFontScaling={true}
          editable={true}
          onChangeText={[Function]}
          placeholder="Username"
          placeholderTextColor="rgba(255,255,255,0.7)"
          style={
            Object {
              "color": "#000",
              "flex": 1,
              "fontSize": 17,
              "height": 50,
              "lineHeight": 24,
              "paddingLeft": 5,
              "paddingRight": 5,
              "top": 1.5,
            }
          }
          underlineColorAndroid="rgba(0,0,0,0)"
          virtual={undefined}
        />
      </View>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        collapsable={undefined}
        hasTVPreferredFocus={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "borderBottomWidth": 1,
            "borderColor": "#D9D5DC",
            "borderLeftWidth": 0,
            "borderRightWidth": 0,
            "borderTopWidth": 0,
            "flexDirection": "row",
            "height": 106.72,
            "marginBottom": 26.68,
            "marginLeft": 15,
            "opacity": 1,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <TextInput
          allowFontScaling={true}
          editable={true}
          onChangeText={[Function]}
          placeholder="Password"
          placeholderTextColor="rgba(255,255,255,0.7)"
          secureTextEntry={true}
          style={
            Object {
              "color": "#000",
              "flex": 1,
              "fontSize": 17,
              "height": 50,
              "lineHeight": 24,
              "paddingLeft": 5,
              "paddingRight": 5,
              "top": 1.5,
            }
          }
          underlineColorAndroid="rgba(0,0,0,0)"
          virtual={undefined}
        />
      </View>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        collapsable={undefined}
        hasTVPreferredFocus={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "alignSelf": "stretch",
            "backgroundColor": "#0277BD",
            "borderBottomWidth": null,
            "borderColor": "#007aff",
            "borderLeftWidth": null,
            "borderRadius": 5,
            "borderRightWidth": null,
            "borderTopWidth": null,
            "elevation": 2,
            "flexDirection": "row",
            "height": 45,
            "justifyContent": "center",
            "marginLeft": 10,
            "marginRight": 10,
            "marginTop": 10,
            "opacity": 1,
            "paddingBottom": 6,
            "paddingTop": 6,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "color": "#fff",
                "fontFamily": "System",
                "fontSize": 16.5,
                "lineHeight": 19,
                "marginLeft": 0,
                "marginRight": 0,
                "paddingLeft": 16,
                "paddingRight": 16,
              },
              Object {
                "textAlign": "center",
              },
            ]
          }
          uppercase={false}
          virtual={undefined}
        >
          Sign In
        </Text>
      </View>
    </View>
  </View>
</RCTSafeAreaView>
`;

exports[`renders correctly with no username 1`] = `
<RCTSafeAreaView
  style={
    Object {
      "backgroundColor": "#03A9F4",
      "flex": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "flex": 1,
          "height": 1334,
        },
        Object {
          "backgroundColor": "#03A9F4",
        },
      ]
    }
    virtual={undefined}
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "alignSelf": "center",
            "flex": 1,
          },
          Object {
            "alignItems": "center",
            "flex": 3,
            "height": 1334,
            "justifyContent": "center",
            "marginBottom": 80.04,
            "marginTop": 80.04,
          },
        ]
      }
      virtual={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": undefined,
              "fontSize": 400.2,
            },
            undefined,
            Object {
              "fontFamily": "zocial",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
          ]
        }
      >
        
      </Text>
    </View>
    <View
      style={
        Array [
          Object {},
          Object {
            "flex": 2,
            "marginBottom": 10,
            "marginLeft": 10,
            "marginRight": 10,
          },
        ]
      }
      virtual={undefined}
    >
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        collapsable={undefined}
        hasTVPreferredFocus={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "borderBottomWidth": 1,
            "borderColor": "#D9D5DC",
            "borderLeftWidth": 0,
            "borderRightWidth": 0,
            "borderTopWidth": 0,
            "flexDirection": "row",
            "height": 106.72,
            "marginBottom": 26.68,
            "marginLeft": 15,
            "opacity": 1,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <TextInput
          allowFontScaling={true}
          editable={true}
          onChangeText={[Function]}
          placeholder="Username"
          placeholderTextColor="rgba(255,255,255,0.7)"
          style={
            Object {
              "color": "#000",
              "flex": 1,
              "fontSize": 17,
              "height": 50,
              "lineHeight": 24,
              "paddingLeft": 5,
              "paddingRight": 5,
              "top": 1.5,
            }
          }
          underlineColorAndroid="rgba(0,0,0,0)"
          virtual={undefined}
        />
      </View>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        collapsable={undefined}
        hasTVPreferredFocus={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "borderBottomWidth": 1,
            "borderColor": "#D9D5DC",
            "borderLeftWidth": 0,
            "borderRightWidth": 0,
            "borderTopWidth": 0,
            "flexDirection": "row",
            "height": 106.72,
            "marginBottom": 26.68,
            "marginLeft": 15,
            "opacity": 1,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <TextInput
          allowFontScaling={true}
          editable={true}
          onChangeText={[Function]}
          placeholder="Password"
          placeholderTextColor="rgba(255,255,255,0.7)"
          secureTextEntry={true}
          style={
            Object {
              "color": "#000",
              "flex": 1,
              "fontSize": 17,
              "height": 50,
              "lineHeight": 24,
              "paddingLeft": 5,
              "paddingRight": 5,
              "top": 1.5,
            }
          }
          underlineColorAndroid="rgba(0,0,0,0)"
          virtual={undefined}
        />
      </View>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        collapsable={undefined}
        hasTVPreferredFocus={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "alignSelf": "stretch",
            "backgroundColor": "#0277BD",
            "borderBottomWidth": null,
            "borderColor": "#007aff",
            "borderLeftWidth": null,
            "borderRadius": 5,
            "borderRightWidth": null,
            "borderTopWidth": null,
            "elevation": 2,
            "flexDirection": "row",
            "height": 45,
            "justifyContent": "center",
            "marginLeft": 10,
            "marginRight": 10,
            "marginTop": 10,
            "opacity": 1,
            "paddingBottom": 6,
            "paddingTop": 6,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "color": "#fff",
                "fontFamily": "System",
                "fontSize": 16.5,
                "lineHeight": 19,
                "marginLeft": 0,
                "marginRight": 0,
                "paddingLeft": 16,
                "paddingRight": 16,
              },
              Object {
                "textAlign": "center",
              },
            ]
          }
          uppercase={false}
          virtual={undefined}
        >
          Sign In
        </Text>
      </View>
    </View>
  </View>
</RCTSafeAreaView>
`;
